RUFAS/biophysical/manure/manure_manager.py

```diff
@@ -755,18 +790,338 @@ def _generate_adjacency_matrix_keys(self) -> list[str]:
+            self._manure_nutrient_manager.nutrients_by_manure_category[manure_type].phosphorus_composition
+        )
+
+        if is_nitrogen_limiting_nutrient:
+            limiting_nutrient_requested_amount = results.nitrogen
+            available_amount_in_pool = \
+                self._manure_nutrient_manager.nutrients_by_manure_category[manure_type].nitrogen
+        else:
+            limiting_nutrient_requested_amount = results.phosphorus
+            available_amount_in_pool = \
+                self._manure_nutrient_manager.nutrients_by_manure_category[manure_type].phosphorus
+
+        proportion_of_limiting_nutrient_to_remove = self._determine_limiting_nutrient_proportion_to_be_removed(
+            limiting_nutrient_requested_amount,
+            available_amount_in_pool
+        )
+        non_limiting_fields = [
+            "water",
+            "ammoniacal_nitrogen",
+            "potassium",
+            "ash",
+            "non_degradable_volatile_solids",
+            "degradable_volatile_solids",
+            "total_solids",
+        ]
+
+        for name, processor in self.all_processors.items():
+            if isinstance(processor, Storage):
+                processor.stored_manure, removal_details = self.compute_stream_after_removal(
+                    stored_manure=processor.stored_manure,
+                    limiting_nutrient_removal_proportion=proportion_of_limiting_nutrient_to_remove,
+                    is_nitrogen_limiting_nutrient=is_nitrogen_limiting_nutrient,
+                    non_limiting_fields=non_limiting_fields
+                )
+                removal_details["manure_type"] = STORAGE_CLASS_TO_TYPE.get(type(processor))
+                self._manure_nutrient_manager.remove_nutrients(removal_details)
+
+    @staticmethod
+    def compute_stream_after_removal(
+        stored_manure: ManureStream,
+        limiting_nutrient_removal_proportion: float,
+        is_nitrogen_limiting_nutrient: bool,
+        non_limiting_fields: list[str],
+    ) -> tuple[ManureStream, dict[str, Any]]:
+        """
+        Returns a new ManureStream with removals applied,
+        plus a dict of how much was removed for each attribute.
+        """
+        if is_nitrogen_limiting_nutrient:
+            limiting = "nitrogen"
+            non_limiting_fields.append("phosphorus")
+        else:
+            limiting = "phosphorus"
+            non_limiting_fields.append("nitrogen")
+
+        removed: dict[str, Any] = {}
+
+        original_limiting_nutrients_in_storage = getattr(stored_manure, limiting)
+        limiting_nutrients_to_remove = original_limiting_nutrients_in_storage * limiting_nutrient_removal_proportion
+        removed[limiting] = limiting_nutrients_to_remove
+
+        updates: dict[str, float] = {limiting: original_limiting_nutrients_in_storage - limiting_nutrients_to_remove}
+
+        for field in non_limiting_fields:
+            original_non_limiting_amount = getattr(stored_manure, field)
```

Comment:
```suggestion
            original_amount = getattr(stored_manure, field)
```
I think things get a little wordy here and maybe within this loop since we know we're only dealing with non-limiting-fields we can simplify the var names?